### ISSUE TITLE
Terminal User Interface equivalent to cockpit

### ISSUE THREAD

**Issue by fischerdouglas** [cockpit-project/cockpit#21000]
To those who needs to deploy and maintain remotely appliances and virtual appliances depending on end user interactions, cockpit is a dream!

A thing that avoid several unnecessary access through vnc/anydesk and etc.

But, there is still a small part that is still a pain.

When network is not available by some reason, cockpit is also unavailable...

So, I imagine that a TUI version of the core of cockpit, ncurses based... With the same goals os the console of VMware ESXi, would be a great thing.

I imagine for example that the same Network Manager the is used on web-gui, will adequately replaced by nm-tui.

So... Is this a this that very far from the objectives of cockpit project?

**Comment by jelly** [comment]
Cockpit has no interest in providing a TUI, this would be a lot of work for the Cockpit team to develop, maintain. If you have a machine without networking you can still access Cockpit via a local browser on localhost.

**Comment by fischerdouglas** [comment]
Thank you for your sincerity!
Considering this, no more to add.

**Comment by barbecued** [comment]
@jelly how do you suggest doing this on a minimal install with no GUI or web browser installed?

**Comment by jelly** [comment]
I would need to know more about your use case but you could setup a bastion host so you only have to run `cockpit-bridge` on the machine itself.

**Comment by fischerdouglas** [comment]
Why not think on a front-end ncurses for cockpit APIs that already exists?

Consume those APIs via loopback...

**Comment by jelly** [comment]
Feel free to give it a try, but it's a lot of work for us and we have no plans in that direction.

**Comment by martinpitt** [comment]
You could search for a JavaScript-capable TUI browser (links and w3m can't do that), but I figure even if one exists, it would look rather bad. But I also don't understand the use case -- if you can get a terminal to a machine (localhost or some remote machine over SSH), you can also get proper Cockpit on it. E.g. https://github.com/flathub/org.cockpit_project.CockpitClient can connect via SSH, or you use "Connect to:" on the login page from a local cockpit package installation, or at worst you do some port forwarding.

> Consume those APIs via loopback...

The APIs are relatively uninteresting -- you can drive most of them (or better CLI equivalents) through plain shell commands. Cockpit's work and knowledge mostly goes into UI design.

So in summary, as @jelly said this isn't something which we are interested in nor capable of doing, that would be an entirely separate project.
